src/server.py

```diff
@@ -1,45 +1,151 @@
+                    else:
+                        response = response_error('404 Not Found')
+                    response = response_ok(body)
+                except SyntaxError:
+                    response = response_error('404 Page Not Found')
+                except NameError:
+                    response = response_error('405 Method Not Allowed')
+                except TypeError:
+                    response = response_error('505 HTTP Version Not Supported')
+                except ValueError:
+                    raise HTTPException(
+                        HTTP_BAD_REQUEST,
+                        "No 'Host :' in header")
+            except SystemError:
+                response = response_error()
+            # full_string += b"\r\n\r\n"
+            # conn.sendall(full_string)
+            conn.sendall(response.encode('utf-8'))
             conn.close()
     except KeyboardInterrupt:
         server.close()
 
 
-def response_ok():
-    """Return HTTP 200 OK message for when the connection is ok."""
-    return (b"HTTP/1.1 200 OK\r\n\r\n"
-            b"Success")
+def response_ok(body_tuple):
+    """Return 200 ok."""
+    first = 'HTTP/1.1 200 OK'
+    second_line = 'Content-Type:' + body_tuple[1] + '; charset=utf-8'
+    date = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header_break = ''
+    body = body_tuple
+    fourth_line = 'Content-Length: {}'.format(len(body))
+    string_list = [first, second_line, date, fourth_line, header_break, body]
+    string_list = '\r\n'.join(string_list) + '\r\n'
+    string_list = string_list.encode('utf-8') + body
+    return string_list
+
+
+def response_error(error='500 Internal Server Error'):
+    """Return 500 internal server error."""
+    first = 'HTTP/1.1 {}'.format(error)
+    second_line = 'Content-Type: text/plain; charset=utf-8'
+    date = email.utils.formatdate(usegmt=True)
+    header_break = ''
+    body = 'The system is down'
+    bytes_ = body.encode('utf-8')
+    fourth_line = 'Content-Length: {}'.format(len(bytes_))
+    string_list = [first, second_line, date, fourth_line, header_break, body]
+    string_list = '\r\n'.join(string_list)
+    return string_list
+
+
+def parse_request(request):
+    """Parse GET request from client, return URI."""
+    print(request)
+    lines = request.split('\r\n\r\n', 1)
+    split_lines = lines[0].split('\r\n', 1)
+    split_request = split_lines[0].split()
+    try:
+        method, uri, proto = split_request
+        print(proto)
+    except ValueError:
+        raise SyntaxError
+    if not method == 'GET':
+        raise NameError
+    elif not proto == 'HTTP/1.1':
+        print('2', proto)
+        raise TypeError
+    if 'Host: ' in split_lines[1]:
+        return uri
+    else:
+        raise ValueError
+
+
+def resolve_uri(uri):
+    """Resolve the uri retrned by parse_uri."""
+    root = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'webroot')
+    full_path = os.path.join(root, uri[1:])
```

Comment:
This is good for stripping the first character from the URI, and you want to do that if the first character is a `/`.  But if it isn't, you might lose valuable information by stripping the first character.  Look at `str.lstrip` instead.